brainfuck/compiler.go

```diff
@@ -0,0 +1,70 @@
+package brainfuck
+
+type Compiler struct {
+	code       string
+	codeLength int
+	position   int
+
+	instructions []*Instruction
+}
+
+func NewCompiler(code string) *Compiler {
```

Comment:
Compiler is public right? Why do you need a `NewCompiler` func which is a constructor?